src/internal/packager/images/pull.go

```diff
@@ -60,50 +63,91 @@ func (i *ImgConfig) PullAll() error {
 		imageMap[src] = img
 	}
 
+	// Create the ImagePath directory
+	err := os.Mkdir(i.ImagesPath, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create image path %s: %w", i.ImagesPath, err)
+	}
+
+	totalBytes := int64(0)
 	for src, img := range imageMap {
 		tag, err := name.NewTag(src, name.WeakValidation)
 		if err != nil {
 			return fmt.Errorf("failed to create tag for image %s: %w", src, err)
 		}
 		tagToImage[tag] = img
+
+		// Get the byte size for this image
+		layers, err := img.Layers()
+		if err != nil {
+			return fmt.Errorf("unable to get layers for image %s: %w", src, err)
+		}
+		for _, layer := range layers {
+			size, err := layer.Size()
+			if err != nil {
+				return fmt.Errorf("unable to get size of layer: %w", err)
+			}
+			totalBytes += size
+		}
 	}
 	spinner.Updatef("Preparing image sources and cache for image pulling")
+	spinner.Success()
 
-	var (
-		progress    = make(chan v1.Update, 200)
-		progressBar *message.ProgressBar
-		title       string
-	)
+	doneSaving := make(chan int)
+	title := fmt.Sprintf("Pulling %d images (%s of %s)", imgCount, utils.ByteFormat(float64(0), 2), utils.ByteFormat(float64(totalBytes), 2))
+	progressBar := message.NewProgressBar(totalBytes, title)
 
+	// Create a thread to update a progress bar as we save the image files to disk
 	go func() {
-		_ = tarball.MultiWriteToFile(i.TarballPath, tagToImage, tarball.WithProgress(progress))
+		for {
+			select {
+			case <-doneSaving:
+				title = fmt.Sprintf("Pulling %d images (%s of %s)", imgCount, utils.ByteFormat(float64(totalBytes), 2), utils.ByteFormat(float64(totalBytes), 2))
+				progressBar.Update(totalBytes, title)
+				progressBar.Successf("Pulling %d images", imgCount)
+				return
+			default:
+				var (
+					dirErr      error
+					currentSize int64
+				)
+				currentSize, dirErr = utils.GetDirSize(i.ImagesPath)
+				if dirErr != nil {
+					message.Warnf("unable to get the updated progress of the image pull: %s", dirErr.Error())
+					time.Sleep(200 * time.Millisecond)
+					continue
+				}
+				title = fmt.Sprintf("Pulling %d images (%s of %s)", imgCount, utils.ByteFormat(float64(currentSize), 2), utils.ByteFormat(float64(totalBytes), 2))
+				progressBar.Update(currentSize, title)
+				time.Sleep(200 * time.Millisecond)
+			}
+		}
 	}()
 
-	for update := range progress {
-		switch {
-		case update.Error != nil && errors.Is(update.Error, io.EOF):
-			progressBar.Successf("Pulling %d images (%s)", len(imageMap), utils.ByteFormat(float64(update.Total), 2))
-			return nil
-		case update.Error != nil && strings.HasPrefix(update.Error.Error(), "archive/tar: missed writing "):
-			// Handle potential image cache corruption with a more helpful error. See L#54 in libexec/src/archive/tar/writer.go
-			message.Warnf("Potential image cache corruption: %s of %v bytes - try clearing cache with \"zarf tools clear-cache\"", update.Error.Error(), update.Total)
```

Comment:
Is image caching gone now?

Reply:
No, it is still being cached, I will add back the handling of this error though.